lib/charms/lte_core_interface/v0/lte_core_interface.py

```diff
@@ -0,0 +1,245 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Library for the `lte-core` relation.
+
+This library contains the Requires and Provides classes for handling the `lte-core`
+interface.
+
+The motivation for creating this library comes from the need of relating charmed EPCs with
+charmed simulated enodeBs and user equipment (UEs), for testing purposes. The interface will share
+the IP address of the MME (Mobility Management Entity) from EPC the to the charm which contains the
+corresponding simulated enodeBs and UEs.
+
+## Getting Started
+From a charm directory, fetch the library using `charmcraft`:
+
+# TODO: publish the library
+```shell
+charmcraft fetch-lib charms.lte_core_interface.v0.lte_core_interface
+```
+
+Add the following libraries to the charm's `requirements.txt` file:
+- jsonschema
+
+### Requirer charm
+The requirer charm is the one requiring to connect to the core
+from another charm that provides this interface.
+
+Example:
+```python
+
+from ops.charm import CharmBase
+from ops.main import main
+
+from charm.lte_core_interface.v0.lte_core_interface import (
+    CoreAvailableEvent,
+    CoreRequires,
+)
+
+
+class DummyCoreRequirerCharm(CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.lte_core_requirer = CoreRequires(self, "lte-core")
+        self.framework.observe(
+            self.lte_core_requirer.on.core_available,
+            self._on_core_available,
+        )
+
+    def _on_core_available(self, event: CoreAvailableEvent):
+        print(event.mme_ipv4_address)
+
+
+if __name__ == "__main__":
+    main(DummyCoreRequirerCharm)
+```
+
+### Provider charm
+The provider charm is the one providing information about the core network
+for another charm that requires this interface.
+
+Example:
+```python
+
+from ops.charm import CharmBase
+from ops.main import main
+
+from charm.lte_core_interface.v0.lte_core_interface import (
+    CoreProvides,
+)
+
+
+class DummyCoreProviderCharm(CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.core_provider = CoreProvides(self, "lte-core")
+        self.framework.observe(
+            self.on.lte_core_relation_joined, self._on_lte_core_relation_joined
+        )
+
+    def _on_lte_core_relation_joined(self, event: RelationJoinedEvent):
+        if self.unit.is_leader():
+            self.core_provider.set_core_information(
+                mme_ipv4_address="mme ipv4 address from the core"
```

Comment:
Is there a reason the IP address can't be passed to the constructor? do we have any extra information by the time `relation_joined` is emitted? the library coudl get the IP address and put it into the relatino data on `relation_joined` internally, giving the client charm/user a better experience. But I might be missing something :)

Reply:
We don't know the IP address at that time.